### PR TITLE
fix: Correct ProfileScan err handling and response

### PR DESCRIPTION
**Pull request by cloudxxx8**
Added more logging and return base response to autodiscovery as well Close https://github.com/edgexfoundry/device-sdk-go/issues/1606

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-sdk-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?)
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

### CODE REVIEW

File: openapi/v3/device-sdk.yaml
Comment (jinlinGuan) [inline openapi/v3/device-sdk.yaml:1]: The `/profilescan` example values are incorrect. Please help to remove the `value` in the response examples.
image

**Review by jinlinGuan — CHANGES_REQUESTED** [review]

File: openapi/v3/device-sdk.yaml
Comment (cloudxxx8) [reply openapi/v3/device-sdk.yaml:1]: updated

Comment (jinlinGuan) [reply openapi/v3/device-sdk.yaml:1]: `value` should also be removed from 404, 409, 423, 500 and 501 response examples.

Comment (cloudxxx8) [reply openapi/v3/device-sdk.yaml:1]: sorry, I just checked and corrected all of them

**Review by jinlinGuan — APPROVED** [review]
LGTM

**Pull request merged**